tests/testthat/test-emissions_profile_any_compute_profile_ranking.R

```diff
@@ -62,9 +62,111 @@ test_that("without crucial columns errors gracefully", {
+  co2 <- tibble(
+    activity_uuid_product_uuid = c("a", "a"),
+    co2_footprint = c(1, 1),
+    ei_activity_name = c("a", "a"),
+    ei_geography = c("a", "a"),
+    isic_4digit = c(NA_character_, "'1375'"),
+    tilt_sector = c("a", "a"),
+    tilt_subsector = c("a", "a"),
+    unit = c("a", "a")
+  )
+  isic <- emissions_profile_any_compute_profile_ranking(co2) |>
+    filter(is.na(isic_4digit) & grouped_by == "isic_4digit")
+
+  unit_isic <- emissions_profile_any_compute_profile_ranking(co2) |>
+    filter(is.na(isic_4digit) & grouped_by == "unit_isic_4digit")
+
+  expect_equal(isic$profile_ranking, NA_integer_)
+  expect_equal(unit_isic$profile_ranking, NA_integer_)
+})
+
+test_that("null `tilt_sector` should be excluded for ranking `tilt_sector` and `unit_tilt_sector` benchmarks", {
+  co2 <- tibble(
+    activity_uuid_product_uuid = c("a", "a"),
+    co2_footprint = c(1, 1),
+    ei_activity_name = c("a", "a"),
+    ei_geography = c("a", "a"),
+    isic_4digit = c("'1375'", "'1375'"),
+    tilt_sector = c(NA_character_, "a"),
+    tilt_subsector = c("a", "a"),
+    unit = c("a", "a")
+  )
+  tilt_sec <- emissions_profile_any_compute_profile_ranking(co2) |>
+    filter(is.na(tilt_sector) & grouped_by == "tilt_sector")
+
+  unit_tilt_sec <- emissions_profile_any_compute_profile_ranking(co2) |>
+    filter(is.na(tilt_sector) & grouped_by == "unit_tilt_sector")
+
+  expect_equal(tilt_sec$profile_ranking, NA_integer_)
+  expect_equal(unit_tilt_sec$profile_ranking, NA_integer_)
+})
+
+test_that("2 and 3 digit ISICs should be excluded for ranking `isic_4digit` and `unit_isic_4digit` benchmarks", {
+  co2 <- tibble(
+    activity_uuid_product_uuid = c("a", "a", "a"),
+    co2_footprint = c(1, 1, 1),
+    ei_activity_name = c("a", "a", "a"),
+    ei_geography = c("a", "a", "a"),
+    isic_4digit = c("'12'", "'123'", "'1234'"),
+    tilt_sector = c("a", "a", "a"),
+    tilt_subsector = c("a", "a", "a"),
+    unit = c("a", "a", "a")
+  )
+  isic <- emissions_profile_any_compute_profile_ranking(co2) |>
+    filter((str_length(isic_4digit) %in% c(4, 5)) & (grouped_by %in% c("isic_4digit", "unit_isic_4digit")))
+  expect_equal(unique(isic$profile_ranking), NA_integer_)
+})
+
+test_that("2 and 3 digit ISICs should be included for ranking
+          all benachmarks other than `isic_4digit` and `unit_isic_4digit` benchmarks", {
+  co2 <- tibble(
+    activity_uuid_product_uuid = c("a", "a", "a"),
+    co2_footprint = c(2, 3, 3),
+    ei_activity_name = c("a", "a", "a"),
+    ei_geography = c("a", "a", "a"),
+    isic_4digit = c("'12'", "'123'", "'1234'"),
+    tilt_sector = c("a", "a", "a"),
+    tilt_subsector = c("a", "a", "a"),
+    unit = c("a", "a", "a")
+  )
+  isic <- emissions_profile_any_compute_profile_ranking(co2) |>
+    filter((str_length(isic_4digit) %in% c(4, 5)) & !(grouped_by %in% c("isic_4digit", "unit_isic_4digit")))
+  expect_false(any(is.na(isic$profile_ranking)))
+})
```

Comment:
Reason to write these these two tests can be understood from this issue: https://github.com/2DegreesInvesting/tiltIndicatorBefore/issues/17